.github/workflows/ci-aarchxx.yml

```diff
@@ -128,26 +128,14 @@ jobs:
           CI_BRANCH: ${{ github.ref }}
         if: ${{ matrix.sve == true && matrix.sve_length != ''}}
 
-      - name: Send failure mail to dynamorio-devs
-        if: failure() && github.ref == 'refs/heads/master'
-        uses: dawidd6/action-send-mail@v2
-        with:
-          server_address: smtp.gmail.com
-          server_port: 465
-          username: ${{secrets.DYNAMORIO_NOTIFICATION_EMAIL_USERNAME}}
-          password: ${{secrets.DYNAMORIO_NOTIFICATION_EMAIL_PASSWORD}}
-          subject: |
-            [${{github.repository}}] ${{github.workflow}} FAILED
-            on ${{github.event_name}} at ${{github.ref}}
-          body: |
-            Github Actions CI workflow run FAILED!
-            Workflow: ${{github.workflow}}/x86-32
-            Repository: ${{github.repository}}
-            Branch ref: ${{github.ref}}
-            SHA: ${{github.sha}}
-            Triggering actor: ${{github.actor}}
-            Triggering event: ${{github.event_name}}
-            Run Id: ${{github.run_id}}
-            See more details on github.com/DynamoRIO/dynamorio/actions/runs/${{github.run_id}}
-          to: dynamorio-devs@googlegroups.com
-          from: Github Action CI
+  send-failure-notification:
+      uses: ./.github/workflows/failure-notification.yml
+      needs: [aarch64-precommit]
+      # We need this to ensure this job runs even if there is failure in
+      # any of the workflows marked as needed.
+      if: ${{ always() }}
+      with:
+        test_suite_status: ${{ format('{0} {1}',
+                                      'aarch64-precommit', needs.aarch64-precommit.result) }}
```

Comment:
Could the job name be a variable to avoid duplicating it here?

Reply:
I checked. Unfortunately there doesn't seem to be one for needs:  https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#needs-context. Despite that I still tried adding a `name` to a job, but nothing shows up if I do needs.<job_id>.name.